### PR TITLE
More resilient network operations in CI

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR merges a few improvements around networking in CI, notably:

- Do not let builds fail if we cannot export Docker layers to the GHA cache. This is a nice to have, and shouldn't fail a complete build.
- Switch back to the Wagon transport for the Maven dependency resolver. [See the original blog post which prompted this](https://questdb.io/blog/maven-troubleshooting-open-source/). We can switch back with the next Maven release, assuming the fix is part of it.

There's also a few clean up commits, happy to split them into a separate PR.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
Started a bunch of runs:

- https://github.com/camunda/zeebe/actions/runs/4670241857
- https://github.com/camunda/zeebe/actions/runs/4670251246
- https://github.com/camunda/zeebe/actions/runs/4670252096
- https://github.com/camunda/zeebe/actions/runs/4670252789
- https://github.com/camunda/zeebe/actions/runs/4670253564
- https://github.com/camunda/zeebe/actions/runs/4670254290
- https://github.com/camunda/zeebe/actions/runs/4670254900

**Comment by npepinpe** [comment]
:facepalm: I had only applied the `ignore-error=true` flag on the `cache-from`, not the `cache-to`, so when exporting it was still acknowledging errors.

I'll start some new runs:

- https://github.com/camunda/zeebe/actions/runs/4670728923
- https://github.com/camunda/zeebe/actions/runs/4670734340
- https://github.com/camunda/zeebe/actions/runs/4670735059
- https://github.com/camunda/zeebe/actions/runs/4670735662
- https://github.com/camunda/zeebe/actions/runs/4670736375
- https://github.com/camunda/zeebe/actions/runs/4670737104

**Comment by npepinpe** [comment]
To be honest, I'm still not 100% sure what the `-U` option is doing. I played around with it locally, and it doesn't always update when I think it should :thinking: 

If a JAR is missing, normal Maven will re-download it. If it's broken though, then `-U` doesn't really seem to solve anything. I think the only useful case is for local development, where your Maven install may cache locally that a dependency does not exist, and will not retry it.

This may then be useful for us if we save this, but I would hope we don't cache the repository on build failures :crossed_fingers: 

So for now I will remove this.

File: .github/actions/build-docker/action.yml [hunk 1]
```diff
@@ -81,8 +81,8 @@ runs:
         tags: ${{ steps.get-image.outputs.result }}
         load: ${{ inputs.push != 'true' }}
         push: ${{ inputs.push }}
-        cache-from: type=gha
-        cache-to: type=gha,mode=max
+        cache-from: type=gha,ignore-error=true
```
Comment (oleschoenburg) [inline .github/actions/build-docker/action.yml:84]: That's a really nice find :+1:

**Review by lenaschoenburg — APPROVED** [review]
Thank you so much for looking into it :pray: 

Changes look good to me :+1: 

bors r+

**Pull request merged**